src/sidebar/store/create-store.js

```diff
@@ -26,28 +26,94 @@ const { createReducer, bindSelectors } = require('./util');
  * @param [any[]] middleware - List of additional Redux middlewares to use.
  */
 function createStore(modules, initArgs = [], middleware = []) {
-  // Create the initial state and state update function.
-  const initialState = Object.assign(
-    {},
-    ...modules.map(m => m.init(...initArgs))
-  );
-  const reducer = createReducer(...modules.map(m => m.update));
+  // Create the initial state and state update function. The "base"
+  // namespace is reserved for non-namespaced modules which will eventually
+  // be converted over.
+
+  // Namespaced objects for initial states.
+  const initialState = {
+    base: null,
+  };
+  // Namespaced reducers from each module.
+  const allReducers = {
+    base: null,
+  };
+  // Namespaced selectors from each module.
+  const allSelectors = {
+    base: {
+      selectors: {},
+      // Tells the bindSelector method to use local state for these selectors
+      // rather than the top level root state.
+      useLocalState: true,
+    },
+  };
+
+  // Temporary list of non-namespaced modules used for createReducer.
+  const baseModules = [];
+
+  // Iterate over each module and prep each module's:
+  //    1. state
+  //    2. reducers
+  //    3. selectors
+  //
+  // Modules that have no namespace get dumped into the "base" namespace.
+  //
+  modules.forEach(module => {
+    if (module.namespace) {
+      initialState[module.namespace] = module.init(...initArgs);
+      allReducers[module.namespace] = createReducer(module.update);
+      allSelectors[module.namespace] = {
+        selectors: module.selectors,
```

Comment:
@robertknight, I intended useLocalState to be a property of the intermediary translation object allSelectors, but not part of any modules (since its temporary). In other words, we would never export that value from any module. It's just hard coded to the "base" set but not any other namespaced module. Do you think that is a issue?

My intention here is that all namespaced module selectors get the top level root state parameter  and would be responsible for drilling down into each namespace they want to access. This allows module selectors to see other states too.

Thoughts on that?